### PR TITLE
Replace director with cluster

### PR DESCRIPTION
**Pull request by sirainen**
DOV-5535. Merge first https://github.com/dovecot/documentation/pull/563 and https://github.com/dovecot/documentation/pull/564. Also fix the architecture png image.

### CODE REVIEW

**Comment by sirainen** [comment]
Merge first https://github.com/dovecot/documentation/pull/563 and https://github.com/dovecot/documentation/pull/564. Also fix the architecture png image.

File: source/admin_manual/dovecot_proxy.rst [hunk 1]
```diff
@@ -4,12 +4,17 @@
...
+ * Authenticate the user
+ * Find out the user's current backend and proxy the connection there
 
 The proxying can be done for all users, or only for some specific users. There are two ways to do the authentication:
 
-1. Forward the password to the remote server. The proxy may or may not perform authentication itself. This requires that the client uses only cleartext authentication mechanism, or alternatively the proxy has access to users' passwords in cleartext.
+#. Forward the password to the remote server. The proxy may or may not perform authentication itself. This requires that the client uses only cleartext authentication mechanism, or alternatively the proxy has access to users' passwords in cleartext.
 
-2. Let Dovecot proxy perform the authentication and login to remote server using the proxy's master password. This allows client to use also non-cleartext authentication mechanism.
+#. Let Dovecot proxy perform the authentication and login to remote server using the proxy's master password. This allows client to use also non-cleartext authentication mechanism.
```
Comment (mrannanj) [inline source/admin_manual/dovecot_proxy.rst:18]: These two lines could be broken smaller too

**Review by mrannanj — CHANGES_REQUESTED** [review]
Surprisingly, `master_user_separator` is also a director setting, and should probably be removed here.

**Comment by cmouse** [comment]
> Surprisingly, `master_user_separator` is also a director setting, and should probably be removed here.

It's not **just** director setting.

**Comment by sirainen** [comment]
`master_user_separator` removal/merging is now in https://github.com/dovecot/documentation/pull/568

File: source/admin_manual/debugging/index.rst [hunk 2]
```diff
@@ -6,7 +6,7 @@ Debugging
 
 Each ``IMAP``, ``POP3`` and ``LMTP`` connection has its own unique session ID.
 This ID is logged in all the lines and passed between Dovecot services, which
-allows tracking it all the way through directors to backends and their various
+allows tracking it all the way through proxies to backends and their various
 processes.
 
 The session IDs look like ``<ggPiljkBBAAAAAAAAAAAAAAAAAAAAAAB>``
```
Comment (cmouse) [inline source/admin_manual/debugging/index.rst:12]: add into .github/actions/spelling/patterns.txt

File: source/admin_manual/cluster/index.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
+.. _dovecot_cluster_architecture:
+
+==============================
+Dovecot Cluster Architecture
+==============================
+
+.. versionadded:: v3.0.0
+
```
Comment (cmouse) [inline source/admin_manual/cluster/index.rst:8]: should mention that this is Dovecot Pro feature.

**Review by cmouse — CHANGES_REQUESTED** [review]

File: source/admin_manual/debugging/index.rst (hunk 2 above)
Comment (sirainen) [reply source/admin_manual/debugging/index.rst:12]: Why does the spell-check pass even without it being there?..

Comment (cmouse) [reply source/admin_manual/debugging/index.rst:12]: dunno, but it would be safer to add it there.

Comment (cmouse) [reply source/admin_manual/debugging/index.rst:12]: ah, it's already there, at least partially...

.github/actions/spelling/patterns.txt:BBAAAAAAAAAAAAAAAAAAAAAAB

Comment (cmouse) [reply source/admin_manual/debugging/index.rst:12]: maybe this is fine.

File: source/configuration_manual/authentication/password_database_extra_fields.rst [hunk 4]
```diff
@@ -52,7 +52,7 @@ specially:
 
   .. versionadded:: v2.2.26+/v2.3
 
-* **forward_<anything>**: In proxy/director, pass the variable to next hop as
+* **forward_<anything>**: In a proxy, pass the variable to the next hop (backend) as
```
Comment (cmouse) [inline source/configuration_manual/authentication/password_database_extra_fields.rst:55]: can we mention the 30 character limit here?

Comment (sirainen) [reply source/configuration_manual/authentication/password_database_extra_fields.rst:55]: Done in https://github.com/dovecot/documentation/pull/571

**Review by cmouse — APPROVED** [review]

**Review by cmouse — APPROVED** [review]

**Pull request merged**